CONTRIBUTING.md

```diff
@@ -40,177 +264,782 @@ since someone else can often do this if you include a complete example. We
+- **Creating new labels:** if you feel a `part:` label is missing, do
+  not hesitate to create it.  If you are not sure, you may discuss it
+  with other contributors and developers on [Gitter][] first.
+- **Closing issues:** if a bug cannot be reproduced anymore, is a
+  duplicate, or should not be considered a bug report in the first
+  place, you should close it.  When doing so, try putting an
+  appropriate `resolved:` label to indicate the reason.  If the bug
+  has been fixed already, and you know in which version, you can add a
+  milestone to it, even a milestone that's already closed, instead of
+  a `resolved:` label.  When closing a duplicate issue, try to add all
+  the additional info that could be gathered to the original issue.
+- **Editing issue titles:** you may want to do so to better reflect
+  the current understanding of the underlying issue.
+- **Editing comments:** feel free to do so to fix typos and formatting
+  only (in particular, some old comments from the Bugzilla era or
+  before are not properly formatted).  You may also want to edit the
+  OP's initial comment (a.k.a. body of the issue) to better reflect
+  the current understanding of the issue, especially if the discussion
+  is long.  If you do so, only add to the original comment, and mark
+  it clearly with an `EDITED by @YourNickname:`.
+- **Hiding comments:** when the discussion has become too long, this
+  can be done to hide irrelevant comments (off-topic, outdated or
+  resolved sub-issues).
+- **Deleting things:** just don't do it.  FWIW, core developers have
+  access to an audit log where all delete actions are listed (among
+  other things).
+- **Pushing a branch or a tag to the main repository:** just don't do
+  it.  We like to reserve the branches on the main repository for the
+  development branch (`master`) and the stable branches (`v...`).  If
+  you push a branch nonetheless, expect it to be deleted promptly and
+  without notice.
+
+Yet to be documented: use of priority, difficulty, `help wanted`, and
+`good first issue` labels, of milestones, of assignments, and of
+GitHub projects.
+
+## Code changes ##
+
+### Using GitHub pull requests ###
+
+If you want to contribute a documentation update, bug fix or feature
+yourself, pull requests (PRs) on the [GitHub
+repository][coq-repository] are the way to contribute directly to the
+Coq implementation (all changes, even the smallest changes from core
+developers, go through PRs).  You will need to create a fork of the
+repository on GitHub and push your changes to a new "topic branch" in
+that fork (instead of using an existing branch name like `master`).
+
+PRs should always target the `master` branch.  Make sure that your
+copy of this branch is up-to-date before starting to do your changes,
+and that there are no conflicts before submitting your PR.  If you
+need to fix conflicts, we generally prefer that you rebase your branch
+on top of `master`, instead of creating a merge commit.
```

Comment:
Very large branches with many commits where merging is much simpler than rebasing (because you fix conflicts only once), very special cases like integration of an external project such as Ltac2...